pilz_teleoperation/src/pilz_teleoperation/ros_message_serializer.py

```diff
@@ -0,0 +1,53 @@
+import genpy
+import rospy
+
+class RosMessageSerializer:
+    """
+    Serialization and De-Serialization of ROS messages for convenient file printing
+    """
+    def __init__(self):
+        self._module_imports = {}
+
+    def write_messages_to_file(self, ros_messages, filename):
+        """
+        Writes a file that can be used with 'import' to restore
+        :param ros_messages: iterable of messages to be serialized.
+                tuple contains the variable name used for serialization and a ROS message object
+        :param filename: target filename, overwritten if existent
+        """
+        with open(filename, 'w') as f:
+            serialized_message = ""
+            for varname, msg in ros_messages:
+                serialized_message += "{} = {}\n".format(varname, self.convert_ros_message_to_python(msg))
+            f.write("\n".join("from {} import {}".format(module, typename) for typename, module in self._module_imports.items()))
+            f.write("\n\n")
+            f.write(serialized_message)
+
+    def convert_ros_message_to_python(self, message, indentation=0):
+        """
+        Serialize a ROS msg into a String Object that can be used with eval() afterwards
+        :rtype: String
+        """
+        if not isinstance(message, (genpy.Message, rospy.rostime.Time, rospy.rostime.Duration)):
+            # end of recursion
+            if isinstance(message, str):
+                return "'{}'".format(message)
+            else:
+                return str(message)
+
+        field_values = []  # ordered list with string-converted attribute values
+        for field_name, field_type in self._get_message_fields(message):
+            val = self.convert_ros_message_to_python(getattr(message, field_name), indentation + 2)
+            field_values.append((field_name,val))
```

Comment:
maybe extract something in separate methods, and thus get rid of the comments

Reply:
aren't comments a good thing?

Reply:
Not necessarily => See Clean Code (Robert C. Martin)
Better if you explain yourself in code, and that can easily be done by extracting some code and use the name of a method to explain what you are doing.

Reply:
e.g.
`field_values = get_ordered_string_converted_attributes(message, indentation)`